content/en/docs/gke/deploy/upgrade.md

```diff
@@ -24,71 +38,167 @@ However, specific upgrades might need manual actions below.
+
+1. Update `./upstream/management` package:
+   ```bash
+   make update
+   ```
+1. Use kpt to set user values:
+   ```bash
+   kpt cfg set -R . name ${NAME}
+   kpt cfg set -R . gcloud.core.project ${PROJECT}
+   kpt cfg set -R . location ${LOCATION}
+   ```
+   Note, you can find out which setters exist in a package and what there current values are by:
+   ```bash
+   kpt cfg list-setters .
+   ```
+1. Apply upgraded config connector:
+   ```bash
+   make apply-kcc
+   ```
+   Note, you can optionally also run `make apply-cluster`, but it should be the same as your existing management cluster.
+1. Check your config connector upgrade is successful:
+   ```bash
+   # For Kubeflow v1.2, it should be 1.29.0
+   $ kubectl get namespace cnrm-system -ojsonpath='{.metadata.annotations.cnrm\.cloud\.google\.com\/version}'
+   1.29.0
+   ```
+
+## Upgrading Kubeflow cluster
+
+**DISCLAIMERS**:
+
+- The upgrade process depends on each Kubeflow application to handle the upgrade properly. There's no guarantee on data completeness unless the application provides such a guarantee.
+- Always recommend backing up your data before an upgrade.
+- Upgrading Kubeflow cluster can be a disruptive process, please schedule some downtime and communicate with your users.
+
+For upgrading from specific versions, there can be manual actions needed, so recommend referring to the specific sections below.
+
+Here are general instructions to upgrade Kubeflow:
+
+1.  The instructions below assume
+
+    - your current working directory is
+
+      ```bash
+      cd ${KF_DIR}
+      ```
+
+    - your kubectl uses a context that connects to your Kubeflow cluster
+      ```bash
+      # List your existing contexts
+      kubectl config get-contexts
+      # Use the context that connects to your Kubeflow cluster
+      kubectl config use-context ${KF_NAME}
+      ```
+
+1.  Edit the Makefile at `./Makefile` and change `MANIFESTS_URL` to point at the version of Kubeflow manifests you want to use
+
+    - Refer to the [kpt docs](https://googlecontainertools.github.io/kpt/reference/pkg/) for more info about supported dependencies
+
+1.  Update the local copies:
+
     ```bash
-    cd ${KF_DIR}/management
+    make update
     ```
 
-1.  Use your management cluster's kubectl context:
+1.  Redeploy:
 
     ```bash
-    # Look at all your contexts
-    kubectl config get-contexts
-    # Select your management cluster's context
-    kubectl config use-context ${MGMTCTXT}
-    # Verify the context connects to the cluster properly
-    kubectl get namespace
+    make apply
     ```
 
-    If you are using a different enviroment, you can always
-    reconfigure the context by:
+    To evaluate the changes before deploying them you can run `make hydrate` and then compare the contents
+    of `.build` to what is currently deployed.
```

Comment:
> compare the contents of `.build` to what is currently deployed.

maybe state clear where/how to check what is currently deployed.?

Reply:
Good point, reworded